chat.go

```diff
@@ -54,50 +54,22 @@ type ChatCompletionRequest struct {
 	FrequencyPenalty float32                 `json:"frequency_penalty,omitempty"`
 	LogitBias        map[string]int          `json:"logit_bias,omitempty"`
 	User             string                  `json:"user,omitempty"`
-	Functions        []*FunctionDefine       `json:"functions,omitempty"`
-	FunctionCall     string                  `json:"function_call,omitempty"`
+	Functions        []*FunctionDefinition   `json:"functions,omitempty"`
+	FunctionCall     any                     `json:"function_call,omitempty"`
 }
 
-type FunctionDefine struct {
+type FunctionDefinition struct {
 	Name        string `json:"name"`
 	Description string `json:"description,omitempty"`
-	// it's required in function call
-	Parameters *FunctionParams `json:"parameters"`
+	// Parameters is an object describing the function.
+	// You can pass a raw byte array describing the schema,
+	// or you can generate the array from a JSONSchema object, using another library.
+	// The only constraint is that the input must implement the JSONEncodable interface.
+	Parameters any `json:"parameters"`
 }
 
-type FunctionParams struct {
```

Comment:
I guess we should keep `FunctionParams` struct to provide a default way to specify function parameters? So in README we would recomment doing `Parameters: &FunctionParams{...}` or using your own `json.RawMessage` (as per @jmacwhyte suggestion).

(See my second point in the earlier [comment](https://github.com/sashabaranov/go-openai/pull/377#issuecomment-1596097407))



cc @vvatanabe @jmacwhyte